Konard/Basic-dnd/B126-Race-Steel-Defender.xml

```diff
@@ -0,0 +1,1217 @@
+			<p>Once your defender uses this feature, you must finish a long rest before it can use it again.</p>
+		</description>
+		<sheet usage="1/Long Rest">
+			<description>If the defender takes damage that would reduce it to 0 hit points, it instead is reduced to 1.</description>
+		</sheet>
+	</element>
+	<element name="Reserve Power" type="Class Feature" source="Basic's Box O' Fixes" id="ID_B126_DEFENDER_UPGRADES_RESERVE_POWER" >
+		<supports>Defender Upgrades</supports>
+		<description>
+			<p>You install a node for conducting magical energy, enchanted to spontaneously repair your steel defender. Whenever the defender uses its Repair ability, you can expend a spell slot to restore an additional 2d8 hit points to it.</p>
+			<p>When you expend a spell slot of 2nd level or higher, the healing increases by 1d8 for each slot level above 1st.</p>
+		</description>
+		<sheet>
+			<description>Whenever the defender uses its Repair ability, you can expend a spell slot to restore an additional 2d8 hit points to it, the healing increases by 1d8 for each slot level above 1st.</description>
+		</sheet>
+	</element>
+	<element name="Riding Harness" type="Class Feature" source="Basic's Box O' Fixes" id="ID_B126_DEFENDER_UPGRADES_RIDING_HARNESS" >
+		<supports>Defender Upgrades</supports>
+		<description>
+			<p>You design and craft a special saddle or harness to aid in using the steel defender as a mount. When an attack targets a creature that is using the defender as a mount, your defender can force the attack to target itself instead. This saddle or harness functions as an exotic saddle.</p>
+		</description>
+		<sheet>
+			<description>When an attack targets a creature that is using the defender as a mount, your defender can force the attack to target itself instead.</description>
+		</sheet>
+	</element>
+	<element name="Runic Engravings" type="Class Feature" source="Basic's Box O' Fixes" id="ID_B126_DEFENDER_UPGRADES_RUNIC_ENGRAVINGS" >
+		<supports>Defender Upgrades</supports>
+		<requirements>[level:6]</requirements>
+		<prerequisite>6th-level artificer</prerequisite>
+		<description>
+			<p><em>Prerequisite: 6th-level artificer</em></p>
+			<p>By inscribing arcane runes onto your steel defender, it is able to utilize some spellcasting abilities. The defender learns a cantrip of your choice from the artificer spell list. It uses your spell attack modifier to hit. The defender can cast this cantrip without any verbal, somatic, or material components. You can select this upgrade multiple times.</p>
+		</description>
+		<sheet>
+			<description>The defender learns a cantrip of your choice from the artificer spell list. It has a spellcasting modifier of {{Arti Int Mod}}.</description>
+		</sheet>
+		<spellcasting name="Artificer" ability="Arti Int Mod" allowReplace="true" >
+			<list known="true">Artificer</list>
+		</spellcasting>
```

Comment:
Since multiple things grant spellcasting it would be better to grant it as a class feature rather than an Upgrade.

Runic Engravings cannot be used along with Spark of Intelligence for example (sheet will not generate)